pkg/services/oauthtoken/oauth_token_util.go

```diff
@@ -0,0 +1,77 @@
+package oauthtoken
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/login/social"
+	"github.com/grafana/grafana/pkg/models"
+	"golang.org/x/oauth2"
+)
+
+var (
+	logger = log.New("oauthtoken")
+)
+
+func GetCurrentOAuthToken(ctx context.Context, user models.SignedInUser) (*oauth2.Token, error) {
+	authInfoQuery := &models.GetAuthInfoQuery{UserId: user.UserId}
+	if err := bus.Dispatch(authInfoQuery); err != nil {
+		logger.Debug("No oauth token for user", "userid", user.UserId, "username", user.Login)
+		// Not necessarily an error.  User may be logged in another way.
+		return nil, nil
```

Comment:
At first, I was returning an error here, but found that when I was logged in as admin with username/password, the Dispatch call returned an error.  Should that be true or am I mistaken?  

If Dispatch returning an error in the non-oauth case is correct, then I think this should remain as-is returning nil, nil instead of propagating the error.

Reply:
Returning `nil, nil` I don't like. As a consumer and Go developer I'm used to checking the err for non-nil and if not expect other returned value is valid. With this you have to check both return values for nil. But I understand why that may be needed here. Returning a special error if user not found and letting the consumer check this is not good either.

Looking at the code handling the GetAuthInfoQuery seems like `models.ErrUserNotFound` is returned as error if no user is found. So I would suggest checking if `err == models.ErrUserNotFound` and if so debug log, otherwise error log. And keeping the `return nil, nil`.

Reply:
I understand checking models.ErrUserNotFound, logging at debug level, and still returning nil, nil in that case.
If there is an error that is not models.ErrUserNotFound, should I log that at error level and return the error (return nil, err)?

Reply:
> If there is an error that is not models.ErrUserNotFound, should I log that at error level and return the error (return nil, err)?

I was thinking log that at error level and return nil, nil to sort of mimic the old behavior. WDYT?

Reply:
The only place where the oauth token was being added in this way previously was in ds_proxy

if proxy.ds.JsonData != nil && proxy.ds.JsonData.Get("oauthPassThru").MustBool() {
	addOAuthPassThruAuth(proxy.ctx, req)
}

with the beginning of the addOAuthPassThruAuth function being:

func addOAuthPassThruAuth(c *models.ReqContext, req *http.Request) {
	authInfoQuery := &models.GetAuthInfoQuery{UserId: c.UserId}
	if err := bus.Dispatch(authInfoQuery); err != nil {
		logger.Error("Error fetching oauth information for user", "userid", c.UserId, "username", c.Login, "error", err)
		return
	}

To maintain the previous status quo, we have to *at some level* ignore *any* error that comes back from the GetAuthInfoQuery call to bus.Dispatch and carry on with sending the request without oauth. An error in that particular location would not stop the request.  This at least applies to the ds_proxy case since it existed previously.  To maintain backward compatibility and consistency, what you propose sounds fine to me.

Reply:
Committed that change to differentiate the log level

Reply:
Would it be possible to get this into the next 7.3.0 beta?

Reply:
Any idea why continuous-integration/drone/pr started failing on this PR?

Reply:
Sorry for slow response. Great. Will have a look. Unfortunately it may be too late to get this into v7.3 stable, but will consider/discuss it.